ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockCreator.java

```diff
@@ -195,8 +203,24 @@ protected BlockCreationResult createBlock(
 
       throwIfStopped();
 
-      final Optional<List<Deposit>> maybeDeposits =
-          Optional.empty(); // TODO 6110: Extract deposits from transaction receipts
+      final DepositsValidator depositsValidator = newProtocolSpec.getDepositsValidator();
+      Optional<List<Deposit>> maybeDeposits = Optional.empty();
+
+      if (depositsValidator instanceof DepositsValidator.AllowedDeposits) {
```

Comment:
Rather than checking the type of the validator, we could enable this based on whether the `depositContractAddress` has a value

Reply:
Sorry @naviechan I gave you a bad suggestion here, this shouldn't be changed. This can't be changed to enable this based on whether `depositContractAddress` as this value will be defined and added to all the network genesis files.

Reply:
No worries, I think its better for the condition to be `depositsValidator instanceof DepositsValidator.AllowedDeposits && depositContractAddress.isPresent()`?

Reply:
Yes that's makes sense, good idea